### PR TITLE
Bigtable: clean up the public surface

### PR DESCRIPTION
**Pull request by igorbernstein2**
* Exclude autogenerated files that aren't used
* ~~Internal implementation details are now package private~~
* Internal implementation details are now @InternalApi
* package-info.java documents the public surface instead of the autogen one
* EnhancedBigtableStubSetting is now public api
* synth.py is responsible for the transformations

### CODE REVIEW

**Comment by kolea2** [comment]
Can/should this be done on the proto level instead?

**Comment by igorbernstein** [comment]
I don’t know of a way. Do you have a recommendation?

**Comment by igorbernstein2** [comment]
Updated PR to only mark things as `InternalApi` instead of marking things as package private, which broke tests using Mockito to mock the stubs

File: google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/BaseBigtableInstanceAdminClient.java [hunk 1]
```diff
@@ -72,79 +73,10 @@
...
- * <pre>
- * <code>
- * BaseBigtableInstanceAdminSettings baseBigtableInstanceAdminSettings =
- *     BaseBigtableInstanceAdminSettings.newBuilder().setEndpoint(myEndpoint).build();
- * BaseBigtableInstanceAdminClient baseBigtableInstanceAdminClient =
- *     BaseBigtableInstanceAdminClient.create(baseBigtableInstanceAdminSettings);
- * </code>
- * </pre>
- */
-@Generated("by gapic-generator")
```
Comment (chingor13) [inline google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/BaseBigtableInstanceAdminClient.java]: Should we preserve this annotation so we mark the class as generated.

Comment (igorbernstein2) [reply google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/BaseBigtableInstanceAdminClient.java]: I was debating this myself. As far as I know nothing uses that annotation and I decided to keep it simple until its needed. If you feel strongly, I can try to retain it

**Comment by igorbernstein2** [comment]
As discussed offline:
- created a gapic-generator feature request: https://github.com/googleapis/gapic-generator/issues/2742
- inserted Autogenerated annotations
- added unit tests to make sure that the InternalApi annotations stick

File: google-cloud-clients/google-cloud-bigtable/synth.py [hunk 2]
```diff
@@ -14,39 +14,112 @@
...
+  s.replace(sources,
+            before=r'^(import (?!static).*?\n)\n',
+            after=r'\1import com.google.api.core.InternalApi;\n\n',
+            flags=re.MULTILINE | re.DOTALL)
 
-s.copy(admin_library / 'gapic-google-cloud-bigtable-admin-v2/src', 'src')
-s.copy(admin_library / 'grpc-google-cloud-bigtable-admin-v2/src', '../../google-api-grpc/grpc-google-cloud-bigtable-admin-v2/src')
-s.copy(admin_library / 'proto-google-cloud-bigtable-admin-v2/src', '../../google-api-grpc/proto-google-cloud-bigtable-admin-v2/src')
+  # Replace javadoc and annotations before every public class with InternalApi
+  # javadoc and annotation
```
Comment (chingor13) [inline google-cloud-clients/google-cloud-bigtable/synth.py]: Can you add the link to the feature request issue for the gapic generator?

Comment (igorbernstein2) [reply google-cloud-clients/google-cloud-bigtable/synth.py]: It's at the top of the file

**Pull request merged**